recipes/google-cloud-cpp/2.x/conanfile.py

```diff
@@ -0,0 +1,336 @@
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+            self.options["protobuf"].shared = True
+            self.options["googleapis"].shared = True
+            self.options["grpc-proto"].shared = True
+            self.options["grpc"].shared = True
+
+    def validate(self):
+        # As-of 2022-03, google-cloud-cpp only supports "Visual Studio >= 2019",
+        # and Visual Studio < 2019 is out of mainline support.
+        # The wikipedia page says this maps to 192* for the MSVC version:
+        #   https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B
+        check_min_vs(self, "192")
+        if is_msvc(self) and self.info.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} shared not supported by Visual Studio")
+
+        if hasattr(self, "settings_build") and cross_building(self):
+            raise ConanInvalidConfiguration(
+                "Recipe not prepared for cross-building (yet)"
+            )
+
+        if (
+            self.settings.compiler == "clang"
+            and Version(self.settings.compiler.version) < "6.0"
+        ):
+            raise ConanInvalidConfiguration("Clang version must be at least 6.0.")
+
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, 14)
+
+        if (
+            self.settings.compiler == "gcc"
+            and Version(self.settings.compiler.version) < "5.4"
+        ):
+            raise ConanInvalidConfiguration("Building requires GCC >= 5.4")
+
+        if self.info.options.shared and \
+           (not self.dependencies["protobuf"].options.shared or \
+            not self.dependencies["googleapis"].options.shared or \
+            not self.dependencies["grpc-proto"].options.shared or \
+            not self.dependencies["grpc"].options.shared):
+            raise ConanInvalidConfiguration(
+                "If built as shared, protobuf, googleapis, grpc-proto, and grpc must be shared as well."
+                " Please, use `protobuf/*:shared=True`, `googleapis/*:shared=True`, `grpc-proto/*:shared=True`,"
+                " and `grpc/*:shared=True`",
+            )
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def requirements(self):
+        self.requires("protobuf/3.21.9", transitive_headers=True)
+        self.requires("grpc/1.50.1", transitive_headers=True)
+        self.requires("nlohmann_json/3.10.0")
+        self.requires("crc32c/1.1.1")
+        self.requires("abseil/20220623.0", transitive_headers=True)
+        self.requires("libcurl/7.88.1")
+        self.requires("openssl/1.1.1t")
```

Comment:
```suggestion
        self.requires("openssl/[>=1.1 <4]")
```

We've updated the `grpc` recipe to use a version range instead, and fixing it here to a version range should hopefully resolve version conflicts (unless anywhere else in the graph is trying to use a specific version, which we can also address).